Fixes for issues #40 and #48

andy-maier opened the pull request:
This PR contains fixes for issues #40 and #48, as proposed in the descriptions.

This includes some restructuring in the processing of distro release files.

Also, the distro release file dict now contains only those entries that were found (as opposed to all
entries with the empty string if no information was found). The users of accessor functions like `id()` or `get*attr()` see no change due to this, because they default non-existing entries to the empty string.

Finally, the distro ID found in the file name of the distro release file is no longer stored in `self.dist`, but as key 'id' in the distro release file dict. This seems to be more consistent, to me.

Note the change in expected test case results, when reviewing.

nir0s commented on ld/__init__.py:
```diff
@@ -69,27 +66,81 @@ def _get_lsb_release_info(self):
+            # is somewhat more tolerant (compared to when searching for the
+            # file), because we want to use what was specified as best as
+            # possible.
+            release_file_pattern = re.compile(r'(\w+)[-_](release|version)')
```
can we move this to globals?

nir0s commented on ld/__init__.py:
```diff
@@ -69,27 +66,81 @@ def _get_lsb_release_info(self):
+            # distro specific files; e.g. CentOS, Oracle, Enterprise all
+            # containing `redhat-release` on top of their own.
+            files.sort()
+            release_file_pattern = re.compile(r'(\w+)[-_]release')
```
and then use it here as well?

andy-maier commented:
I assume you are talking about the `release_file_pattern` variable?

Of course they could become globals, but the reason I removed 'version' from the pattern used for searching the distro release file if not specified, is issue #48. if we think there are distro release files that end with 'version' and where there is no other release information, then we could also have 'version' in the pattern but special-case the `debian_version` case on Ubuntu. However I'd prefer not having 'version' and to wait for people finding a distro that needs support for 'version'.

I just added 'version' again this morning in a private version to double check this, and was wondering why the TestDistRelease did not surface the error of issue #48, and it turns out the reason is that this class does not have a testcase for Ubuntu. I will add one and will update the PR.

I don't see a particular reason for having 'version' in the pattern used for the case where the distro release file was specified by the user. I just did not remove it, because there was no issue similar to #48. I would not object to removing 'version' from this pattern as well.

For now, I'll update the PR with the Ubuntu test case, and I wait for your opinion on making the variables global, and about the use of 'version' in their patterns.

andy-maier commented:
This time, I left the commits separate because the one that reordered the test methods would have convoluted the other changes quite a bit.

On commit eb15b3a:
- I found that adding an Ubuntu testcase for the distro release files does not make sense, because debian_version in ubuntu14 should not really be considered a distro release file. Nevertheless, I have added that file for better testing.
- Because the `slackware-version` file can be considered a distro release file, I added `*version` again to the pattern when the distro release file is searched, and moved the resulting single pattern definition into a global variable. I also moved the pattern for the distro release file content into a global variable.
- I found that I had forgotten to add the support for ignoring certain generic or otherwise excluded distro release file names, and added that now.
The commit message details the changes.
